api/src/opentrons/protocol_api/_types.py

```diff
@@ -16,3 +17,34 @@ class OffDeckType(enum.Enum):
 
 See :ref:`off-deck-location` for details on using ``OFF_DECK`` with :py:obj:`ProtocolContext.move_labware()`.
 """
+
+
+class StagingSlotName(enum.Enum):
+    """Staging slot identifiers."""
+
+    SLOT_A4 = "A4"
+    SLOT_B4 = "B4"
+    SLOT_C4 = "C4"
+    SLOT_D4 = "D4"
+
+    @classmethod
+    def from_primitive(cls, value: str) -> StagingSlotName:
+        str_val = str(value).upper()
```

Comment:
Looks like this can just be

```suggestion
        str_val = value.upper()
```

Since these are never `int`s.